salt/_logging/impl.py

```diff
@@ -233,10 +236,16 @@ def _log(
         stack_info=False,
         stacklevel=1,
         exc_info_on_loglevel=None,
+        once=False,
     ):
         if extra is None:
             extra = {}
 
+        if once:
+            if str(args) in self.ONCECACHE:
+                return
+            self.ONCECACHE.add(str(args))
```

Comment:
This is going to prevent the `str` from being garbage collected which could be a problem for long running processes.

Reply:
Yes, the strings are cached forever. This won't be an endlessly growing cache though. As of now I've only identified one place in the code where this feature is really needed. So, as long as the once option is used sparingly I think it's okay.